README.md

```diff
@@ -67,10 +67,8 @@ Use `conda env update -f <file>` to install all needed dependencies for
 channels:
   - conda-forge
 dependencies:
-  - pip=20
-
   # Datacube
-  - datacube>=1.8.4
+  - datacube>=1.8.5
```

Comment:
> One top level (non-blocking) note: there's a lot of version bumps as a part of this PR, which seem like separate changes from a test infrastructure rebuild. That shouldn't block this PR but it does introduce additional variables to this change set unrelated to the original purpose.

This seems to be a common problem when tweaking test infra. We basically use "frozen" set of dependencies to prevent changes from "outside" causing failures. But we don't have any automated processes to bump those "frozen" sets. As a result when there is a major change this often causes a cascade of changes to make things work again, bumping different libraries, tweaking tests if those bumps cause problems etc.. 

> Do we want to protect against an unexpected minor/major bump, e.g. do datacube~=1.8.5 or datacube~=1.8?

With the new `pip` that now (finally!) attempts to find a consistent set of all packages having loose constraints slows it down a lot, so these version constraints are mostly to help `pip` to resolve things better. Remember this is purely internal test docker, we don't have those constraints on libraries. But for this specific case I will need 1.8.5 datacube for `odc-stac` as I need hashable `GeoBox` class.

Reply:
> Remember this is purely internal test docker, we don't have those constraints on libraries. But for this specific case I will need 1.8.5 datacube for odc-stac as I need hashable GeoBox class.

For sure...this inline comment was for the top-level README in the conda environment instructions, so presumably folks would use this line to set up their own environment for non-test use cases.

I think I flagged this because I saw another project get bit by just doing a "greater than or equal to" dependency recently: https://github.com/dask/s3fs/pull/510. Again, not a big deal, but wanted to flag `~=` as a probably-better pattern than `>=`.